spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsTimeoutException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jms;
+
+/**
+ * A timeout occurred within an async gateway.
+ *
+ * @author Gary Russell
+ * @since 4.3
+ *
+ */
+@SuppressWarnings("serial")
+public class JmsTimeoutException extends RuntimeException {
```

Comment:
Shouldn't we extend here from the `org.springframework.jms.JmsException`?
And yes, I really don't see there an appropriate exception for us...